app/models/vote.rb

```diff
@@ -0,0 +1,16 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :work
+  validates :work_id, presence: true
+  validates :user_id, presence: true,uniqueness: {scope: :work_id, message: "you have already voted for this works"}
+
+  def self.new_vote(user_id:, work_id:)
+    work = Work.find_by(id: work_id)
+    user = User.find_by(id: user_id)
+    if !user || !work
```

Comment:
Style: It's generally preferred to use `x.nil?` over `!x`:

```suggestion
    if user.nil? || work.nil?
```